### ISSUE TITLE
tuning_run appears to mishandle dataframes for the flag parameter

### ISSUE THREAD

**Issue by CMLennon** [rstudio/tfruns#52]
The tuning run function appears to call expand.grid on all data.frames that enter through the `flags` argument.

A quick reproducible example:

a file named 'nnet-unit.R' lives in a directory called 'codedir' - its contents are as follows

```
#flags for tuning_run
FLAGS = flags(
	flag_numeric('width1', 16),
	flag_numeric('dropout', .1)
)

#sequentially build model


model = keras_model_sequential() %>% 
                layer_dense(units = 1, activation = 'linear', input_shape = c(num_zs)) %>%
	        layer_dense(units = FLAGS$width1, input_shape = c(num_zs, activation = 
                              'relu')) %>% layer_dropout(rate = FLAGS$dropout)

#regression problem here so send to linear activation function with 1 unit...
	
model %>% layer_dense(units = 1, activation = 'linear')}



compile(model, optimizer = 'adam', loss = 'mse')

history = model %>% fit(x = train_dat, y =train_lab, epochs = 40, batch_size= 10, verbose = 0)

evaluate(model, test_dat, test_lab, verbose = 0)

k_clear_session()

```

is called from a `tuning_run()` where

`flags = data.frame(dropout = c(0,.1, .15, .2), width1= c(16,32,64,128))`

the final call would be 

`tuning_run(here('codedir', 'nnet-unit.R'), runs_dir = here("tuningdir", 'tunename'), flags = flags,
                confirm = FALSE, echo = FALSE)`

documentation would suggest that the tuning_run would tune over 4 models, but currently it appears to tune over 16. This is also true when there are repeated values in the flags df - that is, if flags were:

`flags = data.frame(dropout = c(0, 0, 0, .1), width = c(32, 64, 128, 32))` tuning_run will still tune over 16 models.

Any guidance as to how to tune over only the models specified in a 'flags' dataframe?

Thank you!

**Comment by dfalbel** [comment]
This works as expected for me with the dev version of tfruns:

```
(use sample parameter to run a random subset)
Training run 1/4 (flags = list(0, 16)) 
```
